ape_trezor/accounts.py

```diff
@@ -4,75 +4,97 @@
+        path = self.data_folder.joinpath(f"{alias}.json")
+        path.write_text(json.dumps(account_data))
+
+    def delete_account(self, alias: str):
+        path = self.data_folder.joinpath(f"{alias}.json")
+
+        if path.exists():
+            path.unlink()
 
 
 class TrezorAccount(AccountAPI):
-    _accountfile: Path
+    _account_file_path: Path
+
+    # Optional because it's lazily loaded
+    _account_client: Optional[TrezorAccountClient] = None
 
     @property
     def alias(self) -> str:
-        return self._accountfile.stem
+        return self._account_file_path.stem
 
     @property
-    def accountfile(self) -> dict:
-        return json.loads(self._accountfile.read_text())
+    def address(self) -> AddressType:
+        return to_address(self.account_file["address"])
 
     @property
-    def hdpath(self) -> str:
-        return self.accountfile["hdpath"]
+    def hdpath(self) -> HDPath:
+        raw_path = self.account_file["hdpath"]
+        return HDPath(raw_path)
 
     @property
-    def client(self):
-        return get_default_client()
+    def account_file(self) -> dict:
+        return json.loads(self._account_file_path.read_text())
 
     @property
-    def address(self) -> AddressType:
-        return to_address(self.accountfile["address"])
+    def _client(self) -> TrezorAccountClient:
+        if self._account_client is None:
+            self._account_client = TrezorAccountClient(self.address, self.hdpath)
+        return self._account_client
 
     def sign_message(self, msg: SignableMessage) -> Optional[MessageSignature]:
-        if msg.version != b"E":
-            return None
-        # TODO: trezor does not support eip712 yet, it only supports eip 191 personal_sign
-        signature = ethereum.sign_message(self.client, parse_hdpath(self.hdpath), msg.body)
-        r = signature.signature[1:33]
-        s = signature.signature[33:65]
-        v = signature.signature[0]
-        return MessageSignature(v, r, s)  # type: ignore
-
-    def sign_transaction(self, txn: TransactionAPI) -> TransactionSignature:
-        # NOTE: Some accounts may not offer signing things
-        vrs = ethereum.sign_tx(
-            self.client,
-            parse_hdpath(self.hdpath),
-            txn.nonce,
-            txn.gas_price,
-            txn.gas_limit,
-            txn.receiver,
-            txn.value,
-            txn.data,
-            txn.chain_id,
-            # tx_type,
-        )
-        return TransactionSignature(vrs[0], vrs[1], vrs[2])  # type: ignore
+        version = _extract_version(msg)
+
+        if version == b"E":
+            vrs = self._client.sign_personal_message(msg.body)
+        elif version == b"0x01":
+            vrs = self._client.sign_typed_data(msg.header, msg.body)
+        else:
+            raise TrezorSigningError(
+                f"Unsupported message-signing specification, (version={version!r})"
+            )
+
+        return MessageSignature(*vrs)  # type: ignore
```

Comment:
Can we verify the type of `r` and `s` here?
If they are `int`, we need to call `to_bytes()` on them (see this PR https://github.com/ApeWorX/ape/pull/366 as a reference).